backend/pkg/connect/get_connectors.go

```diff
@@ -28,7 +29,25 @@ const (
 	connectorStateUnassigned connectorState = "UNASSIGNED"
 	connectorStateRunning    connectorState = "RUNNING"
 	connectorStatePaused     connectorState = "PAUSED"
-	connectorStateFAILED     connectorState = "FAILED"
+	connectorStateFailed     connectorState = "FAILED"
+	connectorStateRestarting connectorState = "RESTARTING"
+)
+
+// connectorStatus is our holistic unified connector status that takes into account not just the
+// connector instance state, but also state of all the tasks within the connector
+type connectorStatus = string
+
+const (
+	// Connector is in "running" state, >0 tasks, all of them running state
+	connectorStatusHealthy connectorStatus = "HEALTHY"
+	// Connector is "error" state
+	connectorStatusUnhealthy connectorStatus = "UNHEALTHY"
+	// Connector is "running" state, 0 tasks OR at least one task in failed state
+	connectorStatusDegraded connectorStatus = "DEGRADED"
+	// Connector is in "paused" state, or at least one task is in paused state
+	connectorStatusPaused connectorStatus = "PAUSED"
```

Comment:
I'd argue if just one task is paused (and not all of them) it's in a degraded state and not "paused" state? As a user I'd expect no data being pushed at all when status == "PAUSED"- @vladoschreiner can chime in again

Reply:
Agree.

We've deliberately simplified the control mechanics to not expose task-level pausing and restarting.

As such, if the connector is PAUSED all tasks should be paused, otherwise it's DEGRADED (something is wrong on a task level).